.circleci/config.yml

```diff
@@ -284,65 +300,76 @@ jobs:
       - setup_vm
       - install_chrome
       - run:
-          name: 'Module Tests (<< parameters.config >>)'
+          name: '⭐ Module Tests (<< parameters.config >>) ⭐'
           command: node build-system/pr-check/module-tests.js --config=<< parameters.config >>
       - store_test_output
       - teardown_vm
   'End-to-End Tests':
     executor:
-      name: amphtml-medium-executor
+      name: docker-medium
     parallelism: 6
     steps:
       - setup_vm
       - install_chrome
       - run:
-          name: 'End-to-End Tests'
+          name: '⭐ End-to-End Tests ⭐'
           command: node build-system/pr-check/e2e-tests.js
       - store_test_output
       - teardown_vm
+  'Safari Tests':
+    executor:
+      name: macos-medium
+    steps:
+      - setup_vm
+      - enable_safari_automation
+      - run:
+          name: '⭐ Safari Tests ⭐'
+          command: node build-system/pr-check/safari-tests.js
```

Comment:
Starting this comment thread to discuss the running time of this job.

1. Doing a full macos build and then running tests (without waiting for the docker nomodule build job was [very slow](https://app.circleci.com/pipelines/github/ampproject/amphtml/14327/workflows/6983c6f3-4a5c-46a5-9846-8154e19ce33f). The build section alone [takes 15 mins](https://user-images.githubusercontent.com/26553114/128743688-61cec969-39f6-49eb-bc77-cea8a35a23af.png).
2. Waiting on the nomodule build and then using those results to run unit / integration / e2e tests on macOS is fairly fast. (We only run a small subset of tests, so sharding doesn't help.)
3. I could further optimize by moving the `amp storybook` and PR-deploy-signal sections out of the nomodule build job and do them in a separate stage. This will allow all testing jobs to start immediately.

Option 1 is a nonstarter. Option 2 is where we stand now, but it adds a minute or two to total round-trip time. Thoughts on option 3?

Reply:
How about we start with 2, and test 3 in a separate PR?

Reply:
Good idea. Follow up SGTM.